Add guideline for secret scanning (#95)
equinor/appsec: docs/guidelines/secret-scanning.md

# Secret Scanning

As developers, we know that secrets like passwords, API keys, and access tokens are critical to our work. But what happens when these secrets accidentally end up in our code, logs or error messages?

> This guideline aim to present relevant tools, some good practices for managing this risk and what to do when we have messed up. The AppSec team provides a 4 hour hands-on workshop to get started with secret scanning.

Reviewer: Some minor grammar things
```suggestion
This guideline aims to present relevant tools, some good practices for managing this risk, and what to do when we have messed up. The AppSec team provides a 4 hour hands-on workshop to get started with secret scanning.
```

- [Recommended tools](#recommended-tools)
- [Where to scan for secrets in our SDLC](#where-to-scan-for-secrets-in-our-sdlc)
- [What to do when we have messed up](#what-to-do-when-we-have-messed-up)
- [How do we manage secrets in our dev environments?](#how-do-we-manage-secrets-in-our-dev-environments)

> ## Recommended tools

Reviewer 2: We have another place for tools as well here - appsec/resources/tools/

May be mention these there as well ? or remove that section  to avoid confusion ?

Author: I'll add a point to these as well.

- Snyk Code (in IDE)
- [Trufflehog](https://github.com/trufflesecurity/trufflehog) (elsewhere)

Please note that:

> - False positives is a sizable challenge, scanners needs to be fine tuned to your context (detectors, testing, verification +++)

Reviewer: ```suggestion
- False positives is a sizeable challenge, scanners need to be fine tuned to your context (detectors, testing, verification +++)
```

- Incremental scans would be helpful, you may not want to scan everything all the time

> - Don't underestimate the cultural change when changing how you work. Document how your team does Secure Development, use in on-boarding, use in retrospectives - especially when you identity "interesting" pattern's.

Reviewer: second part of the sentence could benefit from some reworking

## Where to scan for secrets in our SDLC

> The general idea is to catch secrets in code (and other places) as early as possible. Our recommendation for most teams would be;

Reviewer: I wouldn't use the conditional here, our recommendation either IS or IS NOT. 
Replace ; with :

While your are developing, in your development environment

- Scan in your IDE (using Snyk Code)
- Scan in your local build process (using Trufflehog)
- Scan in git pre-commits (using Trufflehog)

In your CI pipeline

- Scan as part of PR checks (using Trufflehog)
- Scan container images as well (using Trufflehog)

> Other options:

Reviewer: ```suggestion
Other places:
```

- Scanning logs (using Trufflehog)

> ## What to do when we have messed up

Reviewer 2: This section should be highlighted to show important content i.e have a different formatting than the regular sections so its easily distinguishable.

Author: I'll add a few emoji's - let's see how the can make a difference

> We will, or someone - not we, will always mess up. So be prepared, both as a individual developers and as teams!

Reviewer: ```suggestion
We will mess up sooner or later. So be prepared, both as an individual developer and as teams!
```
